src/test/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureServiceSpec.scala

```diff
@@ -106,10 +110,193 @@ class AzureServiceSpec extends AnyFlatSpec with Matchers with ScalaFutures {
+    val svc = new AzureService(crlService, new MockDirectoryDAO(), mockMrgDAO)
+
+    val mockApplication = crlService
+      .buildApplicationManager(
+        managedResourceGroup.managedResourceGroupCoordinates.tenantId,
+        managedResourceGroup.managedResourceGroupCoordinates.subscriptionId
+      )
+      .unsafeRunSync()
+      .applications()
+      .list()
+      .asScala
+      .head
+    when(mockApplication.plan())
+      .thenReturn(new Plan().withName(MockCrlService.defaultManagedAppPlan.name).withPublisher("other publisher"))
+
+    val err = intercept[WorkbenchExceptionWithErrorReport] {
+      svc.createManagedResourceGroup(managedResourceGroup, samRequestContext.copy(samUser = user)).unsafeRunSync()
+    }
+    err.errorReport.statusCode shouldBe Some(StatusCodes.Forbidden)
+    mockMrgDAO.mrgs should not contain managedResourceGroup
+  }
+
+  it should "Forbidden when an Application does not have an accepted Plan publisher" in {
+    val user = Generator.genWorkbenchUserAzure.sample
+    val managedResourceGroup = Generator.genManagedResourceGroup.sample.get
+    val mockMrgDAO = new MockAzureManagedResourceGroupDAO
+    val crlService = MockCrlService(user, managedResourceGroup.managedResourceGroupCoordinates.managedResourceGroupName)
+    val svc = new AzureService(crlService, new MockDirectoryDAO(), mockMrgDAO)
+
+    val mockApplication = crlService
+      .buildApplicationManager(
+        managedResourceGroup.managedResourceGroupCoordinates.tenantId,
+        managedResourceGroup.managedResourceGroupCoordinates.subscriptionId
+      )
+      .unsafeRunSync()
+      .applications()
+      .list()
+      .asScala
+      .head
+    when(mockApplication.plan())
+      .thenReturn(new Plan().withName("other name").withPublisher(MockCrlService.defaultManagedAppPlan.publisher))
+
+    val err = intercept[WorkbenchExceptionWithErrorReport] {
+      svc.createManagedResourceGroup(managedResourceGroup, samRequestContext.copy(samUser = user)).unsafeRunSync()
+    }
+    err.errorReport.statusCode shouldBe Some(StatusCodes.Forbidden)
+    mockMrgDAO.mrgs should not contain managedResourceGroup
+  }
+
+  it should "Forbidden when user is not authorized for Application" in {
+    val user = Generator.genWorkbenchUserAzure.sample
+    val managedResourceGroup = Generator.genManagedResourceGroup.sample.get
+    val mockMrgDAO = new MockAzureManagedResourceGroupDAO
+    val svc =
+      new AzureService(MockCrlService(user, managedResourceGroup.managedResourceGroupCoordinates.managedResourceGroupName), new MockDirectoryDAO(), mockMrgDAO)
+
+    val err = intercept[WorkbenchExceptionWithErrorReport] {
+      svc
+        .createManagedResourceGroup(managedResourceGroup, samRequestContext.copy(samUser = user.map(_.copy(email = WorkbenchEmail("other@email.com")))))
+        .unsafeRunSync()
+    }
+    err.errorReport.statusCode shouldBe Some(StatusCodes.Forbidden)
+    mockMrgDAO.mrgs should not contain managedResourceGroup
```

Comment:
Are there any other sanity checks we could do in this Spec for all the null safe parsing stuff you have in place?

Reply:
I added some